aws/telemetry.ts

```diff
@@ -0,0 +1,239 @@
+                [dashboard.filename()]: dashboard.json,
+              },
+            },
+            {
+              provider: k8sProvider,
+            },
+          );
+        });
+
+        // This is just a list of the dashboard names/configmaps to later configure
+        // the helm chart. It is NOT creating any K8s resources.
+        const dashboardsConfigMaps: Record<string, string> = Object.fromEntries(
+          dashboards.map((d) => [d.name, d.name]),
+        );
+
+        //
+        // For each dashboard/ConfigMap, we then need a dashboard provider.
+        //
+        const dashboardProviders = dashboards.map((dashboard) => {
+          return {
+            name: dashboard.name,
+            orgId: 1,
+            folder: "Lightbend Telemetry",
+            type: "file",
+            disableDeletion: false,
+            editable: true,
+            options: {
+              path: `/var/lib/grafana/dashboards/${dashboard.filename()}`,
+            },
+          };
+        });
+
+        new k8s.helm.v3.Chart(
+          "grafana",
+          {
+            chart: "grafana",
+            fetchOpts: {
+              repo: "https://grafana.github.io/helm-charts",
+            },
+            values: {
+              sidecar: {
+                dashboards: {
+                  enabled: true,
+                  label: "grafana_dashboard",
+                },
+              },
+              // See https://github.com/grafana/helm-charts/blob/main/charts/grafana/README.md#import-dashboards
+              dashboardProviders: {
+                "dashboardproviders.yaml": {
+                  apiVersion: 1,
+                  providers: [...dashboardProviders],
+                },
+              },
+
+              dashboardsConfigMaps: {
+                ...dashboardsConfigMaps,
+              },
+
+              // See https://github.com/grafana/helm-charts/blob/main/charts/grafana/README.md#sidecar-for-datasources
+              datasources: {
+                "datasources.yaml": {
+                  datasources: [
+                    {
+                      name: "Cinnamon Prometheus",
+                      type: "prometheus",
+                      access: "proxy",
+                      url: "http://prometheus-server.default.svc.cluster.local",
+                      editable: true,
+                    },
+                  ],
+                },
+              },
+            },
+          },
+          {
+            provider: k8sProvider,
+          },
+        );
+
+        pulumi.log.info(`Grafana installed with Lightbend Telemetry dashboards.`);
+      })
+      .catch((reason) => {
+        pulumi.log.warn(`Could NOT install Grafana with Lightbend Telemetry dashboards: ${reason}`);
+        pulumi.log.warn("Falling back to a vanilla Grafana installation");
+        this.simpleInstall(k8sProvider);
+      });
+  }
+
+  private simpleInstall(k8sProvider: k8s.Provider): void {
+    new k8s.helm.v3.Chart(
```

Comment:
There is some repetition here, but I guess this is better than trying to be too clever and make the code harder to read.

Reply:
Maybe leave this repetition warning as a comment? When maintaining this code it'll help improve all duplicates.